src/util.coffee

```diff
@@ -11,9 +11,16 @@ class VASTUtil
     @resolveURLTemplates: (URLTemplates, variables = {}) ->
         URLs = []
 
-        variables["ASSETURI"]     = @encodeURIComponentRFC3986(variables["ASSETURI"]) if variables["ASSETURI"]?
-        variables["CACHEBUSTING"] = Math.round(Math.random() * 1.0e+10) unless variables["CACHEBUSTING"]?
-        variables["TIMESTAMP"]    = (new Date).toISOString() unless variables["TIMESTAMP"]?
+        # Encode String variables, when given
+        variables["ASSETURI"]        = @encodeURIComponentRFC3986(variables["ASSETURI"]) if variables["ASSETURI"]?
+        variables["CONTENTPLAYHEAD"] = @encodeURIComponentRFC3986(variables["CONTENTPLAYHEAD"]) if variables["CONTENTPLAYHEAD"]?
+
+        # Set default value for invalid ERRORCODE
+        variables["ERRORCODE"] = 900 if variables["ERRORCODE"]? and not /^[0-9]{3}$/.test(variables["ERRORCODE"])
+
+        # Calc random/time based macros
+        variables["CACHEBUSTING"] = Math.round(Math.random() * 1.0e+8)
```

Comment:
If we really want to comply with the spec, we probably need to left-pad this value with `0`s...  :confounded: